include_core/omrsig.h

```diff
@@ -42,7 +42,9 @@ typedef __sighandler_t sighandler_t;
 typedef void (*sighandler_t)(int sig);
 #elif defined(J9ZOS390) || defined(AIXPPC)
 typedef void (*sighandler_t)(int sig);
+#if !defined(__GNUC__)
 #define __THROW
+#endif
```

Comment:
This is within an `extern "C"` block, so none of the uses of `__THROW` below seem appropriate.
The macro definition, and its uses here and in `omrsig.cpp` should just be removed.

Reply:
It is extern "C" only if `__cplusplus` is defined right?

And if that is the case, the it is safe to remove the define instances below entirely?
L35: `#define __THROW`
Snippet L35:37 
```
#if defined(OSX)
#define __THROW
#endif /* defined(OSX) */
```

And what does that mean for these below where __THROW is being used, does the "__THROW" part need to be removed specifically from every line? https://github.com/eclipse/omr/blob/c256823419b4579cf64f94072337fa108f1761f9/include_core/omrsig.h#L104-L115

Reply:
The definitions of `__THROW` should be removed as well as all uses.

Reply:
I get these errors when removing the specifications:
```
/omr/omrsigcompat/omrsig.cpp:241:1: error: 'signal' is missing exception specification 'throw()'
  241 | signal(int signum, sighandler_t handler)
      | ^
      |                                          throw()
```

This being shown for all 5 functions defined in omrsig.cpp with `__THROW`